sklearn/decomposition/tests/test_pca.py

```diff
@@ -529,6 +529,26 @@ def test_pca_score3():
     assert_true(ll.argmax() == 1)
 
 
+def test_pca_score4():
```

Comment:
> I am going to think about it a bit more and come back to you.

I think a simpler test is 
`assert np.all((self.explained_variance_ - self.noise_variance_) >= 0)`

In terms of eigenvalues of the covariance matrix of X, each of the n_component of the biggest eigenvalues should be bigger than the average of the smaller eigenvalues